### PR TITLE
Polish prepare/ansible documentation

### PR DESCRIPTION
**Pull request by happz**
Includes added pyright cover, two birds with one stone.

Pull Request Checklist

* [x] implement the feature

### CODE REVIEW

**Comment by martinhoyer** [comment]
looks good to me

File: tmt/steps/prepare/ansible.py [hunk 1]
```diff
@@ -64,6 +64,14 @@ class PrepareAnsible(tmt.steps.prepare.PreparePlugin[PrepareAnsibleData]):
     """
     Prepare guest using Ansible.
 
+    Run Ansible playbooks against the guest, by running
+    ``ansible-playbook`` for all given playbooks.
+
+    .. warning::
+
+        When specifying playbooks with paths, all pathts must be
```
Comment (qcheng-redhat) [inline tmt/steps/prepare/ansible.py:72]: typo: pathts -> paths

Comment (happz) [reply tmt/steps/prepare/ansible.py]: Fixed in 91793bcf

File: tmt/steps/prepare/ansible.py [hunk 2]
```diff
@@ -93,13 +108,12 @@ class PrepareAnsible(tmt.steps.prepare.PreparePlugin[PrepareAnsibleData]):
         prepare:
             how: ansible
             playbook:
-              - playbook/one.yml
               - https://foo.bar/two.yml
```
Comment (psss) [inline tmt/steps/prepare/ansible.py:111]:
```suggestion
              - https://foo.bar/one.yml
```

Perhaps?

File: tmt/steps/prepare/ansible.py [hunk 3]
```diff
@@ -93,13 +108,12 @@ class PrepareAnsible(tmt.steps.prepare.PreparePlugin[PrepareAnsibleData]):
...
-              - playbook/three.yml
+              - two.yml
 
-    The playbook path must be relative to the metadata tree root.
+    .. code-block:: shell
 
-    The playbook path should be relative to the metadata tree root.
+        prepare -h ansible --playbook https://foo.bar/two.yml --playbook two.yml
```
Comment (psss) [inline tmt/steps/prepare/ansible.py:116]:
```suggestion
        prepare --how ansible --playbook https://foo.bar/two.yml --playbook two.yml
```

What about rather using long option variants in the examples? To make it obvious to the user at the first sight what the option is about.

**Review by psss — APPROVED** [review]
Looks nice, thanks! Just two nitpicks.

File: tmt/steps/prepare/ansible.py [hunk 4]
```diff
@@ -72,19 +80,26 @@ class PrepareAnsible(tmt.steps.prepare.PreparePlugin[PrepareAnsibleData]):
             how: ansible
             playbook: ansible/packages.yml
 
-    Run multiple playbook in one phase, with extra arguments for
+    .. code-block:: shell
+
+        prepare -h ansible --playbook ansible/packages.yml
```
Comment (happz) [inline tmt/steps/prepare/ansible.py:85]:
```suggestion
        prepare --how ansible --playbook ansible/packages.yml
```

**Comment by qcheng-redhat** [comment]
LGTM.

**Review by qcheng-redhat — APPROVED** [review]

**Review by martinhoyer — APPROVED** [review]
The docs look good to me, just a semi-related question - Shouldn't tmt have ansible as a dependency in some form? afaik, it will only be installed as part of provision-virtual or provision-container optional extras packages. Then there is the PyPI package.  
Maybe at least it should be mentioned in the docs with "make sure you have ansible installed" or something

**Comment by happz** [comment]
> > Shouldn't tmt have ansible as a dependency in some form? afaik, it will only be installed as part of provision-virtual or provision-container optional extras packages.
> 
> Good point! The cleanest way would probably be to require `ansible` from `tmt+prepare-ansible` subpackage as we don't want to install `ansible` with the minimal `tmt` package, right?

And `tmt+finish-ansible`...

**Pull request merged**